### PR TITLE
feat: add feature for trigger

### PR DESCRIPTION
**Pull request by jithin1987**
## Context

Adds a feature for for trigger(rebuild_on) functionality 

## References

https://github.com/screwdriver-cd/screwdriver/issues/768

### CODE REVIEW

File: features/trigger.feature [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        (~commit) -> (build_foo) -> (deploy_foo) -> Foo (build_bar) -> (deploy_bar)
+
+    Rules:
+        - Connected Workflows must be one-directional (no loops)
+        - Configuration for trigger must reside in Workflow/Job being triggered
+        - Triggered workflow/job should start with latest commit on configured SCM branch.
+        - If multiple jobs in a pipeline requires the same external pipeline's Job as trigger, then
+          builds for these jobs should be part of same pipeline event.
+
+    Scenario: Failure
```
Comment (petey) [inline features/trigger.feature]: Can this be a little more descriptive? Like: `Subsequent builds are not triggered after a build failure`

File: features/trigger.feature [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        Given two pipelines PipelineA and PipelineB with following config
+            | job           | requires      |
+            | PipelineB:BAR | PipelineA:FOO |
+        When the "FOO" job in PiplelineA is started
+        And the build failed
+        Then the "BAR" job in PipelineB is not started
+
+    Scenario: Success
```
Comment (petey) [inline features/trigger.feature]: Reverse Trigger: `Builds are triggered after another build is successful`

File: features/trigger.feature [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        Assume Project Bar depends on Project Foo. A user should be able to build Project Bar
+        when Project Foo has been built. A user can also make it specific to a job so that
+        Project Bar -> (build_bar) to be built when Project Foo -> (deploy_foo) is
+        finished successfully.
+
+        Combined Pipeline Workflow
+        (~commit) -> (build_foo) -> (deploy_foo) -> Foo (build_bar) -> (deploy_bar)
+
+    Rules:
+        - Connected Workflows must be one-directional (no loops)
```
Comment (petey) [inline features/trigger.feature]: This is really difficult to determine when there are multiple pipelines involved. I do not think we are detecting a looping case where `A (build_bar) -> B (build_bar) -> A (build_bar)`. Also, it doesn't appear that there is a feature test here specifically about loops.

File: features/trigger.feature [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    Scenario: Success
+        Given two pipelines PipelineA and PipelineB with following config
+            | job           | requires      |
+            | PipelineB:BAR | PipelineA:FOO |
+        When the "FOO" job in PiplelineA is started
+        And the build succeeded
+        Then the "BAR" job in PipelineB is started
+        And that "BAR" build uses the same SHA as latest commit on the branch pipeline is configured
+
+    Scenario: Multiple
```
Comment (petey) [inline features/trigger.feature]: Reverse trigger Fan-out: `Multiple builds triggered in parallel as a result of build success` I wonder if it is worth adding a line that the `BAR` and `HELLO` builds belong to the same event?

File: features/trigger.feature [hunk 5]
```diff
@@ -0,0 +1,54 @@
+@ignore
+Feature: Trigger
```
Comment (petey) [inline features/trigger.feature]: There's actually at least three different features described in this feature file, I'm not sure if you want them all in one or want to break them up: rebuild on/reverse trigger, fan-out, and standard intra-workflow triggers. Will fan-in be added to this, or a different test? The scenarios in this file are mostly dealing with the rebuild on/reverse trigger feature.

Comment (jithin1987) [reply features/trigger.feature]: Fan-in and Fan-out scenarios are different and is already in capture for `workflow`. Fan-out scenario is reemphasized here to capture the fact that there should be only one event. All the scenarios in this feature file arise from the successful completion of a single external pipeline's job.

File: features/trigger.feature [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        Combined Pipeline Workflow
+        (~commit) -> (build_foo) -> (deploy_foo) -> Foo (build_bar) -> (deploy_bar)
+
+    Rules:
+        - Configuration for trigger must reside in Workflow/Job being triggered
+        - Triggered workflow/job should start with latest commit on configured SCM branch.
+        - If multiple jobs in a pipeline requires the same external pipeline's Job as trigger, then
+          builds for these jobs should be part of same pipeline event.
+
+    Scenario: Reverse Trigger. Builds are not triggered if required build is not succesful.
```
Comment (petey) [inline features/trigger.feature]: Minor nit. If you title the feature Reverse Trigger, it doesn't need to be in the scenarios.

**Pull request merged**